app/code/community/PedroTeixeira/Correios/Model/Carrier/CorreiosMethod.php

```diff
@@ -114,6 +114,8 @@ public function collectRates(Mage_Shipping_Model_Rate_Request $request)
 
         $this->_filterMethodByItemRestriction();
         if ($this->_getQuotes()->getError()) {
+            //Fix to allow discounts on delivery areas with restrictions
```

Comment:
No caso, se for remover apenas o if, em localhost tive problemas, não apresentando mais cotação no carrinho.